### PR TITLE
Simplify some redundant code

### PR DESCRIPTION
**Pull request by zhangt2333**
I find some code to refactor using analysis : 
* Redundant `Collection.addAll()` call. It may cause one time array copy.
* The code like `this.messageBytesToRead >= input.readableByteCount() ? input.readableByteCount() : this.messageBytesToRead;` may cause one more time `input.readableByteCount()` call.

### CODE REVIEW

File: spring-expression/src/test/java/org/springframework/expression/spel/ExpressionLanguageScenarioTests.java [hunk 1]
```diff
@@ -96,8 +96,7 @@ public void testScenario_DefiningVariablesThatWillBeAccessibleInExpressions() th
 		// Use the standard evaluation context
 		StandardEvaluationContext ctx = new StandardEvaluationContext();
 		ctx.setVariable("favouriteColour","blue");
-		List<Integer> primes = new ArrayList<>();
-		primes.addAll(Arrays.asList(2,3,5,7,11,13,17));
+		List<Integer> primes = new ArrayList<>(Arrays.asList(2, 3, 5, 7, 11, 13, 17));
```
Comment (sbrannen) [inline spring-expression/src/test/java/org/springframework/expression/spel/ExpressionLanguageScenarioTests.java]: No need to create two lists in this context.

```suggestion
		List<Integer> primes = Arrays.asList(2, 3, 5, 7, 11, 13, 17);
```

File: spring-expression/src/test/java/org/springframework/expression/spel/SpelDocumentationTests.java [hunk 2]
```diff
@@ -420,8 +420,7 @@ public void testVariables() throws Exception {
 	@Test
 	public void testSpecialVariables() throws Exception {
 		// create an array of integers
-		List<Integer> primes = new ArrayList<>();
-		primes.addAll(Arrays.asList(2,3,5,7,11,13,17));
+		List<Integer> primes = new ArrayList<>(Arrays.asList(2, 3, 5, 7, 11, 13, 17));
```
Comment (sbrannen) [inline spring-expression/src/test/java/org/springframework/expression/spel/SpelDocumentationTests.java]: Same as above.

```suggestion
		List<Integer> primes = Arrays.asList(2, 3, 5, 7, 11, 13, 17);
```

**Review by sbrannen — CHANGES_REQUESTED** [review]

**Comment by zhangt2333** [comment]
@sbrannen Aha, Thanks for your code review !

**Comment by sbrannen** [comment]
Thanks for making the requested changes. I plan to merge this after 5.2.4 has been released.

**Comment by sbrannen** [comment]
This has been merged into `master`.

Thanks

**Pull request merged**